### PR TITLE
Enhancement/1901 adsense summary

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #1901

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by ryanwelcher** [comment]
@googlebot I consent.

File: assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+			dateRange: select( CORE_USER ).getDateRange(),
+			metrics,
+		};
+
+		const dailyArgs = {
+			startDate: lastMonth,
+			endDate: now,
+			metrics,
+			dimensions: [ 'DATE' ],
+		};
```
Comment (felixarntz) [inline assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js:78]: Introducing a new dependency here and migrating to pass dates directly goes a bit beyond the scope of this one. Let's instead rely on the `dateRange` slugs that are present in the legacy version of the widget (e.g. `today` and `this-month`). The AdSense module class in PHP has extra logic to support those.

File: assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	}
+
+	const processedData = reduceAdSenseData( daily.rows );
+	const href = getSiteKitAdminURL( 'googlesitekit-module-adsense', {} );
+
+	const currencyHeader = period.headers.find( ( header ) => null !== header.currency && 0 < header.currency.length );
+	const currencyCode = currencyHeader ? currencyHeader.currency : false;
+
+	return (
+		<Layout className="googlesitekit-dashboard-adsense-stats" fill>
```
Comment (felixarntz) [inline assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js:108]: Let's not use this legacy component here, we should be either wrapping this in a `Widget` manually, or alternatively provide `wrapWidget: true` in the registration.

File: assets/js/modules/adsense/datastore/__fixtures__/dashboard-summary-widget-options.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	endDate: '2020-09-11',
+	metrics: [
+		'EARNINGS',
+		'PAGE_VIEWS_RPM',
+		'IMPRESSIONS',
+	],
+	dimensions: [
+		'DATE',
+	],
+};
```
Comment (felixarntz) [inline assets/js/modules/adsense/datastore/__fixtures__/dashboard-summary-widget-options.js:29]: See above, let's update these accordingly to pass `dateRange` for now.

File: assets/js/modules/adsense/index.js [hunk 4]
```diff
@@ -79,3 +82,18 @@ addFilter(
...
 );
+
+domReady( () => {
+	Widgets.registerWidget(
+		'adsenseSummary',
+		{
+			component: DashboardSummaryWidget,
+			width: Widgets.WIDGET_WIDTHS.HALF,
+			priority: 1,
+			wrapWidget: false,
```
Comment (felixarntz) [inline assets/js/modules/adsense/index.js:93]: See above - unless this widget needs to use `Widget` with custom props, it should use `wrapWidget: true` here.

File: package.json [hunk 5]
```diff
@@ -179,6 +180,7 @@
     "jsdom": "^16.3.0",
     "lint-staged": "^10.2.11",
     "mini-css-extract-plugin": "^0.9.0",
+    "moment-locales-webpack-plugin": "^1.2.0",
```
Comment (felixarntz) [inline package.json:183]: See above, let's revert the changes to `package.json` and `package-lock.json` as part of this issue.

**Review by felixarntz — CHANGES_REQUESTED** [review]

File: assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+			daily: store.getReport( dailyArgs ),
+		};
+	} );
+
+	if ( loading ) {
+		return <PreviewBlock width="100%" height="276px" />;
+	}
+
+	if ( error ) {
+		return getDataErrorComponent( 'adsense', error.message );
```
Comment (felixarntz) [inline assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js:94]:
```suggestion
		return getDataErrorComponent( 'adsense', error.message, false, false, false, error );
```

File: assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+	}
+
+	if ( ! today.totals && ! period.totals && ! daily.totals ) {
```
Comment (felixarntz) [inline assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js:97]:
```suggestion
	if ( ! today?.totals && ! period?.totals && ! daily?.totals ) {
```

This may cause a JS error otherwise.

**Review by felixarntz — CHANGES_REQUESTED** [review]

**Comment by felixarntz** [comment]
@ryanwelcher I've quickly addressed a few comments here myself to test further. The rest is mostly LGTM, just a few things that should be improved - particularly, let's not use the `Layout` component for refactored widgets, every widget should be wrapped in a `Widget` component (either manually or via `wrapWidget: true`).

File: assets/js/modules/adsense/index.js (hunk 4 above)
Comment (ryanwelcher) [reply assets/js/modules/adsense/index.js:93]: @felixarntz I have updated this however, I needed to pass false here and use the `Widget` component in order to pass the correct class name.

File: assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js (hunk 2 above)
Comment (ryanwelcher) [reply assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js]: Updated to use `Widget` to be able to pass the class.

**Review by felixarntz — APPROVED** [review]
Fixed two follow up issues, one that was causing a console error, the other one was a visual problem where the new widget had double the padding as intended. Looking good now!
